apache/incubator-zeppelin
# Pluggable resource pools

### What is this PR for?
This is a step towards creating a REST endpoint from resource pools  https://issues.apache.org/jira/browse/ZEPPELIN-699

It allows resources to be saved to disk so that they can be reloaded.

### What type of PR is it?
Improvement

### Todos
* [X] - Add ability to use different kinds of resource pools, and show a second type.

### What is the Jira issue?
Partially implements this:  https://issues.apache.org/jira/browse/ZEPPELIN-699

Note, that this pull request is dependent upon this one:  
https://github.com/apache/incubator-zeppelin/pull/782

### How should this be tested?
1.  Edit an interpreter
2.  Add a new property to the interpreter - the "ResourcePoolClass" - and assign it to "org.apache.zeppelin.resource.VFSResourcePool" 
3.  Run the interpreter

You should see new resource files created in notebook/zeppelin_resources

### Screenshots (if appropriate)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation?  No.
